src/esp/sensor/PinholeCamera.h

```diff
@@ -53,6 +53,14 @@ class PinholeCamera : public VisualSensor {
   virtual Corrade::Containers::Optional<Magnum::Vector2> depthUnprojection()
       const override;
 
+  /**
+   * @brief Draw an observation to the frame buffer using simulator's renderer
+   * @return true if success, otherwise false (e.g., frame buffer is not set)
+   * @param[in] sim Instance of Simulator class for which the observation needs
+   *                to be drawn
+   */
+  virtual bool drawObservationToFramebuffer(sim::Simulator& sim) override;
```

Comment:
I admit I don't use these APIs daily, but after looking around a bit, the distinction between `drawObservation()` and `displayObservation()` seems clear enough already, no need to append `ToFramebuffer` to that. 

This also breaks the consistency -- now there's `drawObservationToFramebuffer()`  which is public and virtual, but `readObservation()` is still protected, which begs for questions like "why is one needed outside but the other is not? is reading not done from a framebuffer?". Additionally, from experience, an API suffixed `ToFramebuffer` implies that I could pass the framebuffer via a parameter to the function, but that's not the case here.

A counterproposal -- rewrite the `Sensor::displayObservation()` virtual to something that's more flexible so we don't need to have another virtual doing mostly the same. Here's an initial patch that should give you the idea:

```patch
diff --git a/src/esp/gfx/RenderTarget.h b/src/esp/gfx/RenderTarget.h
index eb48342b..cbfb4de3 100644
--- a/src/esp/gfx/RenderTarget.h
+++ b/src/esp/gfx/RenderTarget.h
@@ -96,10 +96,13 @@ class RenderTarget {
   void readFrameObjectId(const Magnum::MutableImageView2D& view);
 
   /**
-   * @brief Blits the rgba buffer from internal FBO to default frame buffer
-   * which in case of EmscriptenApplication will be a canvas element.
+   * @brief Blit the RGBA buffer from internal FBO to given frame buffer
+   *
+   * Pass @ref Magnum::GL::defaultFramebuffer to blit to the default
+   * framebuffer. Expects that the framebuffer has the same viewport size as
+   * @ref framebufferSize().
    */
-  void blitRgbaToDefault();
+  void blitRgbaTo(Magnum::GL::AbstractFramebuffer& framebuffer);
 
   // @brief Delete copy Constructor
   RenderTarget(const RenderTarget&) = delete;
diff --git a/src/esp/sensor/PinholeCamera.cpp b/src/esp/sensor/PinholeCamera.cpp
index 1752c27a..a6abacae 100644
--- a/src/esp/sensor/PinholeCamera.cpp
+++ b/src/esp/sensor/PinholeCamera.cpp
@@ -152,13 +152,13 @@ void PinholeCamera::readObservation(Observation& obs) {
   }
 }
 
-bool PinholeCamera::displayObservation(sim::Simulator& sim) {
+bool PinholeCamera::drawObservationTo(sim::Simulator& sim, Mn::GL::AbstractFramebuffer* framebuffer) {
   if (!hasRenderTarget()) {
     return false;
   }
 
   drawObservation(sim);
-  renderTarget().blitRgbaToDefault();
+  if(framebuffer) renderTarget().blitRgbaTo(*framebuffer);
 
   return true;
 }
diff --git a/src/esp/sensor/Sensor.h b/src/esp/sensor/Sensor.h
index 00e2bf95..448b0f2a 100644
--- a/src/esp/sensor/Sensor.h
+++ b/src/esp/sensor/Sensor.h
@@ -104,12 +104,15 @@ class Sensor : public Magnum::SceneGraph::AbstractFeature3D {
   virtual bool getObservationSpace(ObservationSpace& space) = 0;
 
   /**
-   * @brief Display next observation from Simulator on default frame buffer
-   * @param[in] sim Instance of Simulator class for which the observation needs
-   *                to be displayed
+   * @brief Draw next observation from Simulator to given frame buffer
+   * @param sim   Instance of Simulator class for which the observation needs
+   *      to be displayed
+   * @param framebuffer Framebuffer to blit the observation to. If `nullptr`,
+   *    the observation is drawn only to sensor's internal
+   *    @ref gfx::RenderTarget without blitting it anywhere.
    * @return Whether the display process was successful or not
    */
-  virtual bool displayObservation(sim::Simulator& sim) = 0;
+  virtual bool drawObservationTo(sim::Simulator& sim, Magnum::GL::AbstractFramebuffer* framebuffer) = 0;
 
  protected:
   SensorSpec::ptr spec_ = nullptr;
```

This of couse means you need to change all `displayObservation()` call sites to pass the framebuffer as well, but I think that'll pay off in the future, as some platforms (VR is one of them) don't have a concept of a default framebuffer anyway.

Then you also don't need to have an additional `Simulator::drawObservationToFramebuffer()` as it would be handled by the (rewritten) `Simulator::drawObservationTo()` as well.